torchdynamo/variables/tensor.py

```diff
@@ -172,9 +173,14 @@ def create(cls, tx, proxy, example_value=None, nnmodule=None, **options):
             and proxy.node.target == "item"
             and config.capture_scalar_outputs
         ):
-            return UnspecializedPrimitiveVariable.create(
-                tx=tx, proxy=proxy, example_value=torch.tensor(example_value)
-            )
+            if isinstance(example_value, np.number):
+                return UnspecializedNumpyVariable.create(
+                    tx=tx, proxy=proxy, example_value=torch.tensor(example_value)
+                )
+            else:
+                return UnspecializedPythonVariable.create(
+                    tx=tx, proxy=proxy, example_value=torch.tensor(example_value)
+                )
```

Comment:
I think this change comes from https://github.com/pytorch/torchdynamo/pull/417. I can understand we are trying to put ```item``` op into the graph. I revisited and found it's better we add another unspecialized type(which is not reflected in current PR yet), as here we don't need to unwrap the value after graph, but other places need. Adding a property to identify the difference is another way to fix it, but we have to maintain the propagation rules explicitly. Any thoughts? @jansel @voznesenskym

Reply:
We could do something during shape/type propagation in `TensorVariable.create` using example values.